rules/Orm30/Rector/MethodCall/CastDoctrineExprToStringRector.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\Orm30\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\Cast\String_;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Type\ObjectType;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://github.com/doctrine/orm/commit/4d73e3ce7801d3bf3254257332e903d8ecea4096
+ */
+final class CastDoctrineExprToStringRector extends AbstractRector
+{
+    /**
+     * @var array<string>
+     */
+    private array $targetMethods = [
+        'like', 'notLike', 'eq', 'neq', 'lt', 'lte', 'gt', 'gte', 'between',
+        'in', 'notIn', 'isMemberOf', 'isInstanceOf',
+    ];
+
+    /**
+     * @var array<string>
+     */
+    private array $exprFuncMethods = [
+        'lower', 'upper', 'length', 'trim', 'avg', 'max', 'min', 'count',
+        'countDistinct', 'exists', 'all', 'some', 'any', 'not', 'abs', 'sqrt',
+    ];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Casts Doctrine Expr\x to string where necessary.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+                $statements->add(
+                    $builder->expr()->like(
+                        $builder->expr()->lower($column),
+                        $builder->expr()->lower($builder->expr()->literal('%'.$like.'%'))
+                    )
+                );
+                CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+                $statements->add(
+                    $builder->expr()->like(
+                        (string) $builder->expr()->lower($column),
+                        (string) $builder->expr()->lower($builder->expr()->literal('%'.$like.'%'))
+                    )
+                );
+                CODE_SAMPLE
+                )]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof MethodCall) {
+            return null;
+        }
+
+        if (! $this->isObjectType($node->var, new ObjectType('Doctrine\ORM\Query\Expr'))) {
+            return null;
+        }
+
+        if (! in_array($this->getName($node->name), $this->targetMethods, true)) {
+            return null;
+        }
+
```

Comment:
init has changed flag here:

```suggestion
        $hasChanged = false;
```

Reply:
Updated to your suggestion thank you!